### PR TITLE
Add Russian Roulette memories and moods events

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This adds memories to a nearby audience if someone is playing Russian Roulette with a loaded gun.  The person who wins or loses also gains a positive or negative mood event that is long lasting.  All of these events only trigger if the person playing is targeting a lethal area.  

## Why It's Good For The Game

Being able to add Russian Roulette to engravings would be pretty cool.

## Changelog

:cl:
qol: Playing Russian Roulette with lethal intent now creates a mood event for the user and is engrained in any nearby peoples memories. The more bullets the better the memory and mood boost.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
This is a large enough PR that I will ask you to make this without web editing

**Comment by timothymtorres** [comment]
Got a new PC and can now test stuff directly.  Checked with Mothblocks on discord and he is cool with me reopening my PRs that I wasn't able to test.

File: code/__DEFINES/memory_defines.dm [hunk 1]
```diff
@@ -153,4 +153,7 @@
 #define DETAIL_MEDAL_REASON "MEDAL_REASON"
 #define DETAIL_NUKE_CODE "NUKE_CODE"
 
-
+// Russian Roulette
+#define DETAIL_LOADED_ROUNDS "LOADED_ROUNDS"
+#define DETAIL_BODYPART = "BODYPART"
```
Comment (Mothblocks) [inline code/__DEFINES/memory_defines.dm:158]: ?

File: code/__DEFINES/memory_defines.dm [hunk 2]
```diff
@@ -153,4 +153,7 @@
...
+#define DETAIL_OUTCOME = "OUTCOME"
```
Comment (Mothblocks) [inline code/__DEFINES/memory_defines.dm:159]: ?

File: code/__DEFINES/memory_defines.dm (hunk 1 above)
Comment (timothymtorres) [reply code/__DEFINES/memory_defines.dm:158]: https://github.com/tgstation/tgstation/blob/5d16239b5f174417cfccf29c575e5d405c4c3379/strings/memories.json#L422-L432

File: code/__DEFINES/memory_defines.dm (hunk 2 above)
Comment (timothymtorres) [reply code/__DEFINES/memory_defines.dm:159]: https://github.com/tgstation/tgstation/blob/5d16239b5f174417cfccf29c575e5d405c4c3379/strings/memories.json#L422-L432

File: code/__DEFINES/memory_defines.dm (hunk 1 above)
Comment (Mothblocks) [reply code/__DEFINES/memory_defines.dm:158]: No dude your equals sign

File: code/__DEFINES/memory_defines.dm (hunk 2 above)
Comment (Mothblocks) [reply code/__DEFINES/memory_defines.dm:159]: No dude your equals sign

File: code/__DEFINES/memory_defines.dm (hunk 1 above)
Comment (timothymtorres) [reply code/__DEFINES/memory_defines.dm]: Fixes with 6068d8c402016cd0c1adcbf1d139db4d74a079ee

File: code/__DEFINES/memory_defines.dm (hunk 2 above)
Comment (timothymtorres) [reply code/__DEFINES/memory_defines.dm]: Fixed with 6068d8c402016cd0c1adcbf1d139db4d74a079ee

**Comment by timothymtorres** [comment]
This is ready to go!

File: code/datums/mood_events/generic_positive_events.dm [hunk 3]
```diff
@@ -293,3 +293,11 @@
 
 	mood_change *= card_players
 	return ..()
+
+/datum/mood_event/russian_roulette_win
+	description = "I gambled my life and won! I'm lucky to be alive..."
+	mood_change = 2
+	timeout = 5 MINUTES
```
Comment (SuperNovaa41) [inline code/datums/mood_events/generic_positive_events.dm:300]: idk how i feel about the mood buff being so long tbh, you have  a fairly high chance of not dying on the first shot in russian roulette so this just feels like an easy way to get mood

Comment (timothymtorres) [reply code/datums/mood_events/generic_positive_events.dm:300]: There are plenty of other positive events that give around the same mood buff without the possibility of dying.  It's also offset by a severe negative mood if they lose and get revived.

**Comment by timothymtorres** [comment]
Is there something you wanted me to change with this?

**Comment by optimumtact** [comment]
i want you to find a maintainer to open and merge it

**Comment by Mothblocks** [comment]
he found one

**Pull request merged**